### PR TITLE
Update release/maint page for 2.12 release

### PR DESCRIPTION
**Pull request by samccann**
##### SUMMARY

also add better headings so people can find the changelogs

Part of #75883
##### ISSUE TYPE

- Docs Pull Request

##### COMPONENT NAME

docs/docsite/rst/reference_appendices/release_and_maintenance.rst
##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

**Comment by sivel** [comment]
I have some changes in a PR that partially overlap with this PR.  If you want to pull in my changes, specifically the new EOL column, I can just close my PR.  I was planning to merge my PR after the 2.12 release goes out on Nov 8.

https://github.com/ansible/ansible/pull/75940

**Comment by samccann** [comment]
Yep I can bring that column into this PR. I plan on publishing on release day.

**Comment by samccann** [comment]
@sivel - I think I got all your changes incorporated in this PR. Please take a look when you get a chance.

File: docs/docsite/rst/reference_appendices/release_and_maintenance.rst [hunk 1]
```diff
@@ -102,36 +105,43 @@ You can install ``ansible-core`` with ``pip``. See :ref:`intro_installation_guid
...
+`2.12 ansible-core Changelogs`_   Maintained (security **and** general bug fixes)       May 2023
+`2.11 ansible-core Changelogs`_   Maintained (security **and** critical bug fixes)      Nov 2022
+`2.10 ansible-base Changelogs`_   Maintained (security fixes only)                      May 2022
+`2.9 Changelogs`_                 Maintained (pre-collections) (security fixes only)    May 2022
+`2.8 Changelogs`_                 Unmaintained (end of life)                            EOL
+`2.7 Changelogs`_                 Unmaintained (end of life)                            EOL
+`2.6 Changelogs`_                 Unmaintained (end of life)                            EOL
+`2.5 Changelogs`_                 Unmaintained (end of life)                            EOL
+<2.5                              Unmaintained (end of life)                            EOL
+================================= ==================================================== ======================
```
Comment (sivel) [inline docs/docsite/rst/reference_appendices/release_and_maintenance.rst:129]: This LGTM to me

**Comment by lmalivert** [comment]
LGTM @samccann.

**Pull request merged**